CenterForOpenScience/ember-osf-web
# [ENG-4413] Replace BsModal with OsfDialog

-   Ticket: [ENG-4413]
-   Feature flag: n/a

## Purpose

Replace all occurrences of BsModal with OsfDialog and fix any problems that arise.

## Summary of Changes

1. Replace BsModal
2. Fix dashboard timing to keep OsfDialogs from overlapping one another
3. Adjust styling of NewProjectNavigationModal
4. Fix tests
5. Fix the occasional button that wasn't quite ideal

## Side Effects

Some dialogs may look a little different.

## QA Notes

Things to check:
- [ ] Can resubmit to a collection
- [ ] Can confirm an email address
- [ ] Can register a project from the project’s registration page
- [ ] Can delete a fork you own of a project from that forked project’s forks page
- [ ] Can withdraw a registration
- [ ] Can end a registration's embargo early
- [ ] Analytics page Linked nodes dialog still works


[ENG-4413]: https://openscience.atlassian.net/browse/ENG-4413?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
